dynamic/message_factory.go

```diff
@@ -159,22 +154,38 @@ func (r *KnownTypeRegistry) CreateIfKnown(messageName string) proto.Message {
 	}
 }
 
+func wellKnownType(typ reflect.Type) string {
+	v := reflect.New(typ).Elem().Interface()
+	switch m := v.(type) {
+	case interface{ XXX_WellKnownType() string }:
+		return m.XXX_WellKnownType()
+	case proto.Message:
+		mn := proto.MessageName(m)
+		if strings.HasPrefix(mn, "google.protobuf.") &&
```

Comment:
For maximal compatibility, you may want to use the exact same set of messages as before:

https://github.com/golang/protobuf/blob/4c88cc3f1a34ffade77b79abc53335d1e511f25b/protoc-gen-go/generator/generator.go#L1687-L1705

Reply:
Ideally, this logic would be identical to the logic in `protoc-gen-go` which previously generated the `XXX_WellKnownType` method, which uses an explicit list: https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L1687

Reply:
Done.